### PR TITLE
fix: address CodeQL resource lifetime warnings

### PR DESCRIPTION
**Pull request by FrankChen021**
## What changed

- Close test-owned input, output, archive, and JDBC resources with try-with-resources.
- Centralize repeated stream ownership where a downstream component closes the stream.
- Replace the obsolete LGTM annotation and document the few intentional lifecycle-owned resources with precise CodeQL suppressions.

## Why

The current CodeQL snapshot reports 117 resource-lifetime warnings:

- 88 `java/input-resource-leak`
- 16 `java/output-resource-leak`
- 13 `java/database-resource-leak`

Most findings were real test cleanup gaps. The remaining ownership cases outlive the creating method by design and are closed by an enclosing `Closer`, connection, client teardown, or downstream copy operation.

## Impact

This makes resource ownership explicit without changing production behavior. Test cleanup is deterministic, and the intentional long-lived cases are documented at the exact alert sites.

## Root cause

Several tests relied on wrapper or fixture teardown to close nested resources, which CodeQL could not consistently prove. A small number of production and lifecycle tests intentionally transfer ownership beyond the creating method.

## Checks

- 181 focused tests passed
- 2 tests skipped
- `git diff --check`

Created by GPT-5.6-Sol.

### CODE REVIEW

File: sql/src/test/java/org/apache/druid/sql/avatica/DruidAvaticaHandlerTest.java [hunk 1]
```diff
@@ -1051,21 +1055,28 @@ public void testNotTooManyStatementsWhenClosed()
   public void testAutoReconnectOnNoSuchConnection() throws SQLException
   {
     for (int i = 0; i < 50; i++) {
-      final ResultSet resultSet = client.createStatement().executeQuery("SELECT COUNT(*) AS cnt FROM druid.foo");
-      Assert.assertEquals(
-          ImmutableList.of(ImmutableMap.of("cnt", 6L)),
-          getRows(resultSet)
-      );
+      try (Statement statement = client.createStatement()) {
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply sql/src/test/java/org/apache/druid/sql/avatica/DruidAvaticaHandlerTest.java]: Addressed in a4619616f0 by declaring the try-with-resources Statement variable final, consistent with Druid style and the neighboring resource declarations.

[the remark this answers is not in the document]
Comment (FrankChen021) [reply sql/src/test/java/org/apache/druid/sql/avatica/DruidAvaticaHandlerTest.java]: Fixed in a4619616f0: the try-with-resources Statement is now declared final, matching the surrounding code and project style.

**Comment by FrankChen021** [comment]
Addressed the latest Copilot findings in commit 39c3a1d864. ServerWrapper is now AutoCloseable, the max-frame, min-frame, and async Avatica tests close it on every exit path, their executors shut down in finally blocks, and the redundant nested CompressionUtils output-stream resource was removed. Targeted tests passed (3 Avatica tests and 1 CompressionUtils test), as did full processing/sql validation with Checkstyle and PMD.

**Comment by FrankChen021** [comment]
Also addressed the two earlier suppressed Copilot suggestions in commit e04124afae: the white-rabbit fixture now has an explicit missing-resource diagnostic, and the HttpShuffleClient mock is typed through InputStreamResponseHandler so openSegmentFile returns ListenableFuture<InputStream> instead of ListenableFuture<Object>. CompressionUtilsTest (1 targeted test) and HttpShuffleClientTest (5 tests) passed; full processing/indexing-service validation with Checkstyle and PMD also passed.

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 19 of 19 changed files.

---

This is an automated review by Codex GPT-5.6-Sol

**Review by GWphua — APPROVED** [review]
